### PR TITLE
POL-57-Update to Budget Alerts Policy

### PR DESCRIPTION
**Pull request by gopinathmahakud1**
POL-57-Update to Budget Alerts Policy

### Description

Update to Budget Alerts Policy. A graph added to represent the budget alerts

### Issues Resolved

Policy has been enhanced. Newly added a graph to represent the budget alerts.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

https://governance.rightscale.com/org/78/projects/30598/policy-incidents/5dd66e3cca13c90001342532

image

### CODE REVIEW

File: cost/budget_alerts/budget_alert.pt [hunk 1]
```diff
@@ -346,24 +453,27 @@ escalation "esc_budget_alert" do
...
+  validate $filtered_target do
     summary_template "Budget Exceeded"
     # ignore changes to the monthly spend and runrate, since those will change constantly
     hash_exclude "actual", "runrate", "total"
     detail_template <<-EOS
 # Budget Exceeded
 
 ### Cost Metric: {{ parameters.param_cost_metric }}
-### Budget Alert Type: {{ parameters.param_type }}
+### Budget Alert Type: {{ parameters.param_type }} Budget Exceeded
```
Comment (cdwilhelm) [inline cost/budget_alerts/budget_alert.pt]: remove budget exceeded here.  the param_type should be in the summary_template above.

File: cost/budget_alerts/budget_alert.pt [hunk 2]
```diff
@@ -346,24 +453,27 @@ escalation "esc_budget_alert" do
 end
 
 policy "budget_alert" do
-  validate_each $filtered_target do
+  validate $filtered_target do
     summary_template "Budget Exceeded"
```
[the remark this answers is not in the document]
Comment (gopinathmahakud1) [reply cost/budget_alerts/budget_alert.pt]: I have addressed this. Please review

File: cost/budget_alerts/budget_alert.pt [hunk 3]
```diff
@@ -346,24 +453,27 @@ escalation "esc_budget_alert" do
...
+### Budget Alert Type: {{ parameters.param_type }} Budget Exceeded
+### Monthly Spend for {{ data.monthYear }}
+
+![Spending Overview Chart](https://image-charts.com/chart?{{ data.chartType }}&{{ data.chartData }}&{{ data.chartSize }}&{{ data.chartTitle }}&{{ data.chartImage }}&{{ data.chartColor }}&{{ data.chartLebel }}&{{ data.chartYAxisLebel }}&{{ data.chartYAxis }}&{{ data.chartLebelPosition }}&{{data.chartDataAutoScale}}&{{data.chartDataValue}} "Actual v. Monthly Cost Report")
 
-| Billing Center Name | Budget | Current Monthly Spend | Monthly Runrate | Billing Center ID |
-| ------------------- | ------ | --------------------- | --------------- | ----------------- |
-{{ range data -}}
-| {{ .name }} | {{.budget}} | {{.actual}} | {{.runrate}} | {{.id}} | 
+| Billing Center Name | Budget | Month to Date Spend | Forecasted Full Month Spend | Over Budget Amount | Over Budget Percent | Billing Center ID |
```
[the remark this answers is not in the document]
Comment (gopinathmahakud1) [reply cost/budget_alerts/budget_alert.pt]: I have addressed this. Please review

File: cost/budget_alerts/budget_alert.pt (hunk 1 above)
Comment (gopinathmahakud1) [reply cost/budget_alerts/budget_alert.pt]: I have addressed this. Please review

**Pull request merged**